x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/LifecycleLicenseIT.java

```diff
@@ -0,0 +1,130 @@
+import static org.elasticsearch.xpack.TimeSeriesRestDriver.rolloverMaxOneDocCondition;
+import static org.hamcrest.CoreMatchers.containsStringIgnoringCase;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+
+public class LifecycleLicenseIT extends ESRestTestCase {
+
+    private String policy;
+    private String dataStream;
+
+    @Before
+    public void refreshDatastream() {
+        dataStream = "logs-" + randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        policy = "policy-" + randomAlphaOfLength(5);
+    }
+
+    @After
+    public void resetLicenseToTrial() throws Exception {
+        putTrialLicense();
+        checkCurrentLicenseIs("trial");
+    }
+
+    public void testCreatePolicyUsingActionAndNonCompliantLicense() throws Exception {
+        String snapshotRepo = randomAlphaOfLengthBetween(4, 10);
+        createSnapshotRepo(client(), snapshotRepo, randomBoolean());
+
+        assertOK(client().performRequest(new Request("DELETE", "/_license")));
+        checkCurrentLicenseIs("basic");
+
+        ResponseException exception = expectThrows(ResponseException.class,
+            () -> createNewSingletonPolicy(client(), policy, "cold", new SearchableSnapshotAction(snapshotRepo, true)));
+        assertThat(EntityUtils.toString(exception.getResponse().getEntity()),
+            containsStringIgnoringCase("policy [" + policy + "] defines the [" + SearchableSnapshotAction.NAME + "] action but the " +
+                "current license is non-compliant for [searchable-snapshots]"));
+    }
+
+    public void testSearchableSnapshotActionErrorsOnInvalidLicense() throws Exception {
+        String snapshotRepo = randomAlphaOfLengthBetween(4, 10);
+        createSnapshotRepo(client(), snapshotRepo, randomBoolean());
+        createNewSingletonPolicy(client(), policy, "cold", new SearchableSnapshotAction(snapshotRepo, true));
+
+        createComposableTemplate(client(), "template-name", dataStream,
+            new Template(Settings.builder().put(LifecycleSettings.LIFECYCLE_NAME, policy).build(), null, null));
+
+        assertOK(client().performRequest(new Request("DELETE", "/_license")));
+        checkCurrentLicenseIs("basic");
+
+        indexDocument(client(), dataStream, true);
+
+        // rolling over the data stream so we can apply the searchable snapshot policy to a backing index that's not the write index
+        rolloverMaxOneDocCondition(client(), dataStream);
+
+        String backingIndexName = DataStream.getDefaultBackingIndexName(dataStream, 1L);
+        // the searchable_snapshot action should start failing (and retrying) due to invalid license
+        assertBusy(() -> {
+            Map<String, Object> explainIndex = explainIndex(client(), backingIndexName);
+            assertThat(explainIndex.get("action"), is(SearchableSnapshotAction.NAME));
+            assertThat((Integer) explainIndex.get("failed_step_retry_count"), greaterThanOrEqualTo(1));
```

Comment:
can we also check the actual failure here? So that we know that the action failed because of a license exception? Just sanity double checking, since we also test that everything is working after license is restored.